include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -158,7 +158,7 @@ __pattern_walk3(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _ForwardIt
     oneapi::dpl::__par_backend_hetero::__parallel_for(_BackendTag{}, ::std::forward<_ExecutionPolicy>(__exec),
                                                       unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n,
                                                       __buf1.all_view(), __buf2.all_view(), __buf3.all_view())
-        .wait();
+        .__deferrable_wait();
```

Comment:
similar comment to `__pattern_walk1`.

Reply:
@danhoeflinger please explain which chain of call exactly you mean.

Reply:
UPD: all tests passed on `CPU` and `GPU` devices and no `assert` has been occurred,
I mean `assert` from
```C++
    template <typename _T>
    constexpr auto
    __wait_and_get_value(const _T& __val)
    {
        wait();

        assert(__get_waited_for_result() && "Result was not waited");

        return __val;
    }
```
So we may have hope that the current state of the code is correct.

Reply:
We discussed and think its best to leave this for now prior to release, and perhaps make more changes after release to switch some patterns to a system like `__pattern_walk2`.